app/controllers/samples_controller.rb

```diff
@@ -991,8 +991,13 @@ def contigs_fasta
     contigs_fasta_s3_path = pr.contigs_fasta_s3_path
 
     if contigs_fasta_s3_path
-      @contigs_fasta = get_s3_file(contigs_fasta_s3_path)
-      send_data @contigs_fasta, filename: @sample.name + '_contigs.fasta'
+      filename = @sample.name + '_contigs.fasta'
+      @contigs_fasta_url = get_presigned_s3_url(contigs_fasta_s3_path, filename)
+      if @contigs_fasta_url
+        redirect_to @contigs_fasta_url
+      else
+        send_data nil, filename: filename
```

Comment:
If `get_presigned_s3_url()` returns `nil`, the user will get a 0 byte file download, is that correct? Would it be better to render an error instead?

Reply:
at least we should log the error

Reply:
Good call. Changed to render and log an error instead.